multicluster/controllers/multicluster/internal/leader_elector.go

```diff
@@ -0,0 +1,105 @@
+package internal
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"antrea.io/antrea/multicluster/controllers/multicluster/common"
+)
+
+/**
+ * Lets consider a ClusterSet with 2 leader clusters and 1 member cluster
```

Comment:
We discussed this topic before. I am not very convinced we should go active/active leaders, and each leader computes imported resources for its electors (BTW, if each member just chooses its own leader independently, should it be even called "election"?). If we go this way, does it mean all leaders might duplicate the same computation for different member clusters? Do you see a possibility multiple members can share imported resources (for example, if one Service should be imported by 10 members, do we compute 10 times, and create 10 imported resources)?

Reply:
Actually maybe its just my bad, maybe my comment is incorrect. Leader is not computing import resource per member, there is only one import resource that all members will import from the elected-leader. I changed the comment now. I think the only reason leader needs to know which member it is leader for is I think visibility. Im thinking maybe we need to do some cleanup action.  E.g. if at some point, L1 was the leader but M1 changed to L2 as leader, if M1 had written some status for import-resources with either success/error, then maybe L1 should do a cleanup job to clear those.

Reply:
Ah, that sounds much better. I still feel we might consider a model with only one active elected leader, and all members just write/watch from that leader. But I am fine to start with the current active/active design too, if we already have most code for that. Yes, no matter which way we go we should handle leader change, and re-sync and cleanup after that.

Reply:
Sure, thank you. I will take it up in a future patch.